packages/components/src/form-token-field/index.tsx

```diff
@@ -0,0 +1,696 @@
+	}
+
+	function handleCommaKey() {
+		if ( inputHasValidValue() ) {
+			addNewToken( incompleteTokenValue );
+		}
+
+		return true; // PreventDefault.
+	}
+
+	function moveInputToIndex( index: number ) {
+		setInputOffsetFromEnd( value.length - Math.max( index, -1 ) - 1 );
+	}
+
+	function moveInputBeforePreviousToken() {
+		setInputOffsetFromEnd( ( prevInputOffsetFromEnd ) => {
+			return Math.min( prevInputOffsetFromEnd + 1, value.length );
+		} );
+	}
+
+	function moveInputAfterNextToken() {
+		setInputOffsetFromEnd( ( prevInputOffsetFromEnd ) => {
+			return Math.max( prevInputOffsetFromEnd - 1, 0 );
+		} );
+	}
+
+	function deleteTokenBeforeInput() {
+		const index = getIndexOfInput() - 1;
+
+		if ( index > -1 ) {
+			deleteToken( value[ index ] );
+		}
+	}
+
+	function deleteTokenAfterInput() {
+		const index = getIndexOfInput();
+
+		if ( index < value.length ) {
+			deleteToken( value[ index ] );
+			// Update input offset since it's the offset from the last token.
+			moveInputToIndex( index );
+		}
+	}
+
+	function addCurrentToken() {
+		let preventDefault = false;
+		const selectedSuggestion = getSelectedSuggestion();
+
+		if ( selectedSuggestion ) {
+			addNewToken( selectedSuggestion );
+			preventDefault = true;
+		} else if ( inputHasValidValue() ) {
+			addNewToken( incompleteTokenValue );
+			preventDefault = true;
+		}
+
+		return preventDefault;
+	}
+
+	function addNewTokens( tokens: string[] ) {
+		const tokensToAdd = uniq(
+			tokens
+				.map( saveTransform )
+				.filter( Boolean )
+				.filter( ( token ) => ! valueContainsToken( token ) )
+		);
+
+		if ( tokensToAdd.length > 0 ) {
+			const newValue = clone( value );
+			newValue.splice( getIndexOfInput(), 0, ...tokensToAdd );
+			onChange( newValue );
+		}
+	}
+
+	function addNewToken( token: string ) {
+		if ( ! __experimentalValidateInput( token ) ) {
+			speak( messages.__experimentalInvalid, 'assertive' );
+			return;
+		}
+		addNewTokens( [ token ] );
+		speak( messages.added, 'assertive' );
+
+		setIncompleteTokenValue( '' );
+		setSelectedSuggestionIndex( -1 );
+		setSelectedSuggestionScroll( false );
+		setIsExpanded( ! __experimentalExpandOnFocus );
+
+		if ( isActive ) {
+			focus();
+		}
+	}
+
+	function deleteToken( token: string | TokenItem ) {
+		const newTokens = value.filter( ( item ) => {
+			return getTokenValue( item ) !== getTokenValue( token );
+		} );
+		onChange( newTokens );
+		speak( messages.removed, 'assertive' );
+	}
+
+	function getTokenValue( token: { value: string } | string ) {
+		if ( 'object' === typeof token ) {
+			return token.value;
+		}
+
+		return token;
+	}
+
+	function getMatchingSuggestions(
+		searchValue = incompleteTokenValue,
+		_suggestions = suggestions,
+		_value = value,
+		_maxSuggestions = maxSuggestions,
+		_saveTransform = saveTransform
+	) {
+		let match = _saveTransform( searchValue );
+		const startsWithMatch: string[] = [];
+		const containsMatch: string[] = [];
+		const selected = _value.map( ( item ) => {
```

Comment:
Would something like `normalizedValue` be a better name for this variable?

```suggestion
		const normalizedValue = _value.map( ( item ) => {
```